gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/service/MinioHandler.java

```diff
@@ -191,6 +192,17 @@ private boolean isRunning(ProcessFileMinio processFileMinio) {
         return false;
     }
 
+    void checkIfAFileWithSameTypeAlreadyExist(ProcessFile processFile, List<TaskWithStatusUpdate> listTaskWithStatusUpdate) {
```

Comment:
Technically, this method does not only check if a file with same type already exists in the waitingFiles map, it also takes into account the timestamps and it removes this file from the map. The name of this method could be adapted to reflect this behaviour. For example: `removeFileWithSameTypeAndTimestampsFromWaitingFiles` (maybe the name is a bit long, if you see a better alternative, go for it).

Reply:
Understood